dora/shell/src/main/java/alluxio/cli/fsadmin/report/SummaryCommand.java

```diff
@@ -88,101 +70,26 @@ private void printMetaMasterInfo() throws IOException {
-        print(zkAddress);
-      }
-      mIndentationLevel--;
-    }
-
-    if (masterInfo.getRaftJournal()) {
-      print("Raft-based Journal: true");
-      print("Raft Journal Addresses: ");
-      mIndentationLevel++;
-      for (String raftAddress : masterInfo.getRaftAddressList()) {
-        print(raftAddress);
-      }
-      mIndentationLevel--;
-    } else {
-      print("Raft-based Journal: false");
-    }
-    String formatString = "%-32s %-8s %-32s";
-    print(String.format(formatString, "Master Address", "State", "Version"));
-    for (MasterVersion masterVersion: masterInfo.getMasterVersionsList()) {
-      NetAddress address = masterVersion.getAddresses();
-      print(String.format(formatString,
-              address.getHost() + ":" + address.getRpcPort(),
-          masterVersion.getState(),
-          masterVersion.getVersion()));
-    }
-  }
-
-  /**
-   * Prints Alluxio block master information.
-   */
-  private void printBlockMasterInfo() throws IOException {
     Set<BlockMasterInfoField> blockMasterInfoFilter = new HashSet<>(Arrays
         .asList(BlockMasterInfoField.LIVE_WORKER_NUM, BlockMasterInfoField.LOST_WORKER_NUM,
             BlockMasterInfoField.CAPACITY_BYTES, BlockMasterInfoField.USED_BYTES,
             BlockMasterInfoField.FREE_BYTES, BlockMasterInfoField.CAPACITY_BYTES_ON_TIERS,
             BlockMasterInfoField.USED_BYTES_ON_TIERS));
     BlockMasterInfo blockMasterInfo = mBlockMasterClient.getBlockMasterInfo(blockMasterInfoFilter);
 
-    print("Live Workers: " + blockMasterInfo.getLiveWorkerNum());
-    print("Lost Workers: " + blockMasterInfo.getLostWorkerNum());
-
-    print("Total Capacity: "
-        + FormatUtils.getSizeFromBytes(blockMasterInfo.getCapacityBytes()));
-
-    mIndentationLevel++;
-    Map<String, Long> totalCapacityOnTiers = new TreeMap<>((a, b)
-        -> (FileSystemAdminShellUtils.compareTierNames(a, b)));
-    totalCapacityOnTiers.putAll(blockMasterInfo.getCapacityBytesOnTiers());
-    for (Map.Entry<String, Long> capacityBytesTier : totalCapacityOnTiers.entrySet()) {
-      print("Tier: " + capacityBytesTier.getKey()
-          + "  Size: " + FormatUtils.getSizeFromBytes(capacityBytesTier.getValue()));
-    }
-
-    mIndentationLevel--;
-    print("Used Capacity: "
-        + FormatUtils.getSizeFromBytes(blockMasterInfo.getUsedBytes()));
-
-    mIndentationLevel++;
-    Map<String, Long> usedCapacityOnTiers = new TreeMap<>((a, b)
-        -> (FileSystemAdminShellUtils.compareTierNames(a, b)));
-    usedCapacityOnTiers.putAll(blockMasterInfo.getUsedBytesOnTiers());
-    for (Map.Entry<String, Long> usedBytesTier: usedCapacityOnTiers.entrySet()) {
-      print("Tier: " + usedBytesTier.getKey()
-          + "  Size: " + FormatUtils.getSizeFromBytes(usedBytesTier.getValue()));
+    ObjectMapper objectMapper = new ObjectMapper();
+    SummaryOutput summaryInfo = new SummaryOutput(masterInfo, blockMasterInfo, mDateFormatPattern);
```

Comment:
I think your `JobServiceOutput` can follow this pattern, get the info by RPC and pass it to the output object purely for  display. 

And even if you cannot get certain info from RPC, you can still continue to display with broken information like:
```
BlockMasterInfo blockMasterInfo = null;
try {
  blockMasterInfo = mBlockMasterClient.getBlockMasterInfo(blockMasterInfoFilter);
} catch (Exception e) {
  // log the exception and actually you can try to continue, if that's necessary
}
SummaryOutput summaryInfo = new SummaryOutput(blockMasterInfo, ...)
```
And in `SummaryOutput` when it tries to display, if certain info is null you can simply display something like "Failed to connect to master for Block/Worker information" and skip that part.

Actually you may choose to implement this in a follow-up PR (don't have to finish all that before merging).